### PR TITLE
Doc/redefine store orga

### PR DESCRIPTION
**Pull request by matthiasbeyer**
As proposed in personal discussion with @TheNeikos

### CODE REVIEW

**Comment by neithernut** [comment]
ACK.

File: doc/src/02000-store.md [hunk 1]
```diff
@@ -71,11 +71,50 @@ So if the store exists in `/home/user/store/`, a file with the Storepath
...
+
+is
+
+    "imag://module/some/sub/folder/example"
+
+As shown in the example, a link to imag-internal entries, the link is prefixed
+with a "imag://" identifier.
+A link to external content MUST NEVER be prefixed this way.
+The path of the internal link MUST NEVER be relative, but always absolute from
+the the root directory of the store.
```
Comment (TheNeikos) [inline doc/src/02000-store.md]: "the the"

**Comment by matthiasbeyer** [comment]
Ready for cleanup.

**Comment by TheNeikos** [comment]
There is one part that might bite us in the butt.

Semver allows identifiers with a dash, akin to `2.3-alpha` how do we handle that?

**Comment by matthiasbeyer** [comment]
@TheNeikos Simply allow it. I mean... why not? The version information is a postfix, so there isn't anything wrong with that, is there? AFAIK, the first character of a semver is always a number...

**Comment by neithernut** [comment]
@matthiasbeyer then what about `blink-182-2.0`?

**Comment by TheNeikos** [comment]
What @neithernut said. I think we should use something else than a dash. A possible one would be `~`.

So we would just search for the last one and evverything after that is the version.

**Comment by matthiasbeyer** [comment]
@TheNeikos Sounds reasonable. I'm not sure what characters are allowed by FS, so we can argue here on which character would fit. A tilde sounds good to me, though.

We have to restrict the interface so that a filename in the store cannot end with this character, then. This should go into the paper. Tell me where.

**Comment by TheNeikos** [comment]
A tilde is an allowed character under btrfs, not sure about ntfs tho

**Comment by matthiasbeyer** [comment]
[Tidle is in 7-bit-ascii](http://www.neurophys.wisc.edu/comp/docs/ascii/#table) and I bet [all major FS allow 7-bit ascii as filenames](https://en.wikipedia.org/wiki/Filename#Comparison_of_filename_limitations).

**Comment by neithernut** [comment]
A tilde is also used by some editors and office suites for backups/swap files. Even Windows supports a tilde in filenames (not that I care).

**Comment by TheNeikos** [comment]
ACK from me

**Comment by neithernut** [comment]
ACK from me, too.

**Comment by matthiasbeyer** [comment]
Squashed into two commits.

@homu r+

**Comment by matthiasbeyer** [comment]
@homu r+

**Pull request merged**